### PR TITLE
WIP POC Get timestamp from rest api

### PR DESCRIPTION
**Pull request by meravingen123**
Signed-off-by: Serhii Lozynskyi <ext-serhii.lozynskyi@here.com>

### CODE REVIEW

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 1]
```diff
@@ -280,7 +281,7 @@ class AUTHENTICATION_API AuthenticationClient {
    * completed.
    */
   using SignInClientResponse =
-      client::ApiResponse<SignInResult, AuthenticationError>;
```
Comment (andescu) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h]: This is an API change, we cannot do it so easily.

File: olp-cpp-sdk-authentication/src/AuthenticationClient.cpp [hunk 2]
```diff
@@ -19,16 +19,16 @@
...
 #include <rapidjson/document.h>
 #include <rapidjson/istreamwrapper.h>
 #include <rapidjson/stringbuffer.h>
 #include <rapidjson/writer.h>
+
 #include <boost/uuid/uuid.hpp>
 #include <boost/uuid/uuid_generators.hpp>
 #include <boost/uuid/uuid_io.hpp>
+#include <chrono>
+#include <sstream>
```
Comment (andescu) [inline olp-cpp-sdk-authentication/src/AuthenticationClient.cpp]: Boost is not a system library.
Please move system includes back to where they were before.

File: olp-cpp-sdk-authentication/src/AuthenticationClient.cpp [hunk 3]
```diff
@@ -659,6 +626,39 @@ std::string AuthenticationClient::Impl::base64Encode(
   return ret;
 }
 
+ServerTimeResponse AuthenticationClient::Impl::GetServerTime(
+    std::string server_url, const std::shared_ptr<http::Network>& network,
```
Comment (andescu) [inline olp-cpp-sdk-authentication/src/AuthenticationClient.cpp]:
```suggestion
   std::string server_url, http::Network& network,
```
No need for shared_ptr reference.

File: olp-cpp-sdk-authentication/src/AuthenticationClient.cpp [hunk 4]
```diff
@@ -659,6 +626,39 @@ std::string AuthenticationClient::Impl::base64Encode(
...
+    const http::NetworkSettings& network_settings,
+    client::CancellationContext& context) {
+  server_url.append(kTimestampEndpoint);
+  http::NetworkRequest request(server_url);
```
Comment (andescu) [inline olp-cpp-sdk-authentication/src/AuthenticationClient.cpp]:
```suggestion
  http::NetworkRequest request(std::move(server_url));
```

File: olp-cpp-sdk-authentication/src/AuthenticationClient.cpp [hunk 5]
```diff
@@ -250,262 +270,209 @@ AuthenticationClient::Impl::Impl(const std::string& authentication_server_url,
...
+  auto context = client::TaskContext::Create(task, callback);
+  pending_requests->Insert(context);
+  ExecuteOrSchedule(task_scheduler, [=]() {
+    context.Execute();
+    pending_requests->Remove(context);
+  });
+  return context.CancelToken();
+}
+
+client::HttpResponse BlockedNetworkSend(
```
Comment (andescu) [inline olp-cpp-sdk-authentication/src/AuthenticationClient.cpp]: I would not call it ```BlockedNetworkSend```.
Rather ```NetworkSendSync``` or simple ```NetworkSend``` as it is obvious to be blocking.

**Comment by r0busta** [comment]
Is this PR still open for reviews?

**Pull request closed without merging**